plugins/ua_nodestore_binary_encoded.c

```diff
@@ -0,0 +1,1163 @@
+viewNodeDecode(const UA_ByteString *src, size_t *offset, UA_ViewNode* viewNode) {
+    UA_StatusCode retval = UA_STATUSCODE_GOOD;
+    retval |= UA_Byte_decodeBinary(src, offset, &viewNode->eventNotifier);
+    retval |= UA_Boolean_decodeBinary(src, offset, &viewNode->containsNoLoops);
+    return retval;
+}
+
+
+
+static void
+readRowLookuptable(char ltRow [], int length, lookUpTable *lt, int row) {
+    /**
+     * There are four contents per row
+     * |IdentifierType|Identifier|nodePosition|nodeSize|
+     * Each data is separated by space
+     */
+    size_t index = 0;
+    int column = 1;
+    char ltTemp[MAX_ROW_LENGTH] = {0};
+    for (int j = 0 ; j < length; j++) {
+        if(ltRow[j] != ' ') {
+            ltTemp[index] = ltRow[j];
+            index++;
+        }
+        if(ltRow[j] == ' ' || j == length - 1) {
+            switch(column) {
+            case 1:
+                if(strtol(ltTemp, NULL, 10) == UA_NODEIDTYPE_NUMERIC) {
+                    lt[row].nodeId.identifierType = UA_NODEIDTYPE_NUMERIC;
+                }
+                if(strtol(ltTemp, NULL, 10) == UA_NODEIDTYPE_STRING) {
+                    lt[row].nodeId.identifierType = UA_NODEIDTYPE_STRING;
+                }
+                break;
+            case 2:
+                if(lt[row].nodeId.identifierType == UA_NODEIDTYPE_NUMERIC) {
+                    lt[row].nodeId.identifier.numeric = (UA_UInt32)strtol(ltTemp, NULL, 10);
+                }
+                if(lt[row].nodeId.identifierType == UA_NODEIDTYPE_STRING) {
+                    lt[row].nodeId.identifier.string = UA_STRING_NULL;
+                    lt[row].nodeId.identifier.string.length = index;
+                    ltTemp[index] = '\0';
+                    lt[row].nodeId.identifier.string = UA_String_fromChars(&ltTemp[0]);
+                }
+                break;
+            case 3:
+                lt[row].nodePosition = (size_t) strtol(ltTemp, NULL, 10);
+                break;
+            case 4:
+                lt[row].nodeSize = (size_t) strtol(ltTemp, NULL, 10);
+                break;
+            default:
+                break;
+            }
+            /* Clear the array contents */
+            for(int i = 0; i < MAX_ROW_LENGTH; i++) {
+                ltTemp[i] = 0;
+            }
+            column++;
+            index = 0;
+        }
+    }
+}
+
+static lookUpTable*
+UA_Lookuptable_Initialize(UA_UInt32 *ltSize, const char *const path) {
+    int ch;
+    UA_UInt32 nodeCount = 0; // To count the number of nodes
+    FILE *fpLookuptable;
+    fpLookuptable = fopen(path, "r");
+    if(!fpLookuptable) {
+        UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "The opening of file lookupTable.bin failed");
+    }
+    while((ch = fgetc(fpLookuptable)) != EOF) {
+        if(ch == '\n') {
+            nodeCount++;
+        }
+    }
+    fclose(fpLookuptable);
+    *ltSize = nodeCount;
+    lookUpTable *lt = (lookUpTable *)UA_calloc(*ltSize, sizeof(lookUpTable));
```

Comment:
Check for NULL every time memory is allocated.
And handle the error of course. We are in memory constrained environments.

This method might be simpler if you just pass in the pointer to `lookupTable`.